src/api/Api.js

```diff
@@ -229,14 +229,35 @@ class Api {
             requestUrl = requestUrl.substr(0, requestUrl.indexOf('?'));
         }
 
+        // Encode existing query parameters, since tomcat does not accept unencoded brackets. Throw
+        // an error if they're already encoded to prevent double encoding.
+        if (query) {
+            let decodedURL;
+
+            try {
+                decodedURL = decodeURIComponent(query);
+            } catch (err) {
+                return Promise.reject(new Error('Query parameters in URL are invalid'));
+            }
+
+            const isEncoded = query !== decodedURL;
+
+            if (isEncoded) {
+                return Promise.reject(
+                    new Error('Cannot process URL-encoded URLs, pass an unencoded URL'),
```

Comment:
This is quite likely a conscious and sound decision, but I just wanted to point out that to test if the query was encoded upon entry, the query is decoded. So in theory you could argue that instead of throwing this error, you could do (at line 251):
```javascript
query = customEncodeURIComponent(isEncoded ? decodedURL : query);
```
The advantage of throwing an error is that you don't have to worry about weird edge cases, like strings with double encoding....

Reply:
> The advantage of throwing an error is that you don't have to worry about weird edge cases, like strings with double encoding

Yeah exactly, that was my reasoning here. It's probably quite unlikely, but I thought it'd be the only way I could be sure that the string that's passed as url definitely isn't encoded.